datalad/cmd.py

```diff
@@ -1091,7 +1091,12 @@ def _get_bundled_path():
             alongside = False
         else:
             annex_path = op.dirname(op.realpath(annex_fpath))
-            alongside = op.lexists(op.join(annex_path, 'git'))
+            bundled_git_path = op.join(annex_path, 'git')
+            # we only need to consider bundled git if it's actually different
+            # from default. (see issue #5030)
+            alongside = op.lexists(bundled_git_path) and \
+                        bundled_git_path != find_executable('git')
```

Comment:
Okay, ignoring the guards, that expands to 
```
join(dirname(realpath(find_executable('git-annex'))), 'git') == find_executable('git')
```

Is the `realpath` on the left but not right a concern?